src/lib/components/common/HighlightGroup.svelte

```diff
@@ -0,0 +1,72 @@
+<!-- @component A collection of highlights from search results.-->
+
+<script lang="ts">
+  import { _ } from "svelte-i18n";
+
+  type H = $$Generic;
+
+  export let open = false;
+  export let highlights: [string, H][] = [];
```

Comment:
@eyeseast Here's an instance of using generic types with component props. Because I define `H` as generic, it will infer the entry value type of `highlights` based on what I pass in. That means that when I pass `highlight` back to the slot, the type is the same as it was given to the component.

This allows us to create a common wrapper for both page highlights and note highlights, while deferring their actual rendering to the parent. You'll see what I mean in a second.

Reply:
Interesting. I'm not sure I totally get it, and [this RFC](https://github.com/dummdidumm/rfcs/blob/ts-typedefs-within-svelte-components/text/ts-typing-props-slots-events.md#discarded-alternative-1) makes it look like `$$Generics` might not last, but I'm willing to try it and adapt as we need to.

Reply:
Ah thanks for that link—looks like this is the better way to implement:

```
<script lang="ts" generics="H">
```